sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Expression.scala

```diff
@@ -230,12 +230,24 @@ abstract class UnaryExpression extends Expression with trees.UnaryNode[Expressio
     val eval = child.gen(ctx)
     // reuse the previous isNull
     ev.isNull = eval.isNull
-    eval.code + s"""
-      ${ctx.javaType(dataType)} ${ev.primitive} = ${ctx.defaultValue(dataType)};
-      if (!${ev.isNull}) {
-        ${ev.primitive} = ${f(eval.primitive)};
-      }
-    """
+    child match {
+      case Literal(null, _) =>
```

Comment:
I'd like to fix this in Literal.genCode, or you have do fix it in many places, for example, BinaryExpression also should fix it.